File: backend-api/internal/models/monster.go
package models

type Monster struct {
	ID string `json:"id"`

	DetailURL string `json:"detail_url"`

	Image *string `json:"image"`

	Name string `json:"name"`

	Race *string `json:"race"`

	Element *string `json:"element"`

	Size *string `json:"size"`

	Location *string `json:"location"`

	Level *int `json:"level"`

	HP *string `json:"hp"`

	BaseEXP *string `json:"base_exp"`

	JobEXP *string `json:"job_exp"`
}

type MonsterDetail struct {
	ID string `json:"id"`

	DetailURL string `json:"detail_url"`

	Image *string `json:"image"`

	Name string `json:"name"`

	Race *string `json:"race"`

	Element *string `json:"element"`

	Size *string `json:"size"`

	Location *string `json:"location"`

	Level *int `json:"level"`

	HP *string `json:"hp"`

	BaseEXP *string `json:"base_exp"`

	JobEXP *string `json:"job_exp"`

	STR *int `json:"str"`

	AGI *int `json:"agi"`

	VIT *int `json:"vit"`

	INTStat *int `json:"int_stat"`

	DEX *int `json:"dex"`

	LUK *int `json:"luk"`

	ATK *string `json:"atk"`

	MATK *string `json:"matk"`

	DEF *string `json:"def"`

	MDEF *string `json:"mdef"`

	HIT *string `json:"hit"`

	FLEE *string `json:"flee"`

	MoveSpeed *string `json:"move_speed"`

	ASPD *string `json:"aspd"`

	RawJSON *string `json:"raw_json"`

	RawHTML *string `json:"raw_html"`
}

File: archive-romhandbook-ui/lib/types/Monster.ts
export type Monster = {

    id: string

    detail_url: string

    image: string | null

    name: string

    race: string | null

    element: string | null

    size: string | null

    location: string | null

    level: number | null

    hp: string | null

    base_exp: string | null

    job_exp: string | null
}

export type MonsterDetail = Monster & {

    str: number | null

    agi: number | null

    vit: number | null

    int_stat: number | null

    dex: number | null

    luk: number | null

    atk: string | null

    matk: string | null

    def: string | null

    mdef: string | null

    hit: string | null

    flee: string | null

    move_speed: string | null

    aspd: string | null

    raw_json: string | null

    raw_html: string | null
}

export type ApiResponse<T> = {

    success: boolean

    data: T

    meta: unknown
}

export type PaginatedApiResponse<T> = {

    success: boolean

    data: T[]

    meta: {
        page: number
        limit: number
        total: number
        has_next: boolean
    }
}